### ISSUE TITLE
Change `dim` to `axis` for neural network classification ops to match others for the TF 1.0.0 API?

### ISSUE THREAD

**Issue by beckermr** [tensorflow/tensorflow#7391]
It looks like the neural network classification ops still use `dim` whereas the rest of the API is being moved to `axis` in v1.0.0. Is this purposeful, or was this bit just missed?

**Comment by beckermr** [comment]
<img width="1284" alt="screen shot 2017-02-09 at 10 56 02 am" src="https://cloud.githubusercontent.com/assets/5296416/22794074/70921e78-eeb7-11e6-914c-61977aee574b.png">

**Comment by girving** [comment]
@aselle Looks like we missed one.  I imagine this is too late for 1.0, but maybe we could add the *axis* parameter as an alternative name?

**Comment by aselle** [comment]
Yes, for most of the axis changes (except reverse and concat) we actually just kept the old name at the end of the argument list so both work. We can do the same here.

**Comment by beckermr** [comment]
Yay! Thank you!

**Comment by yongtang** [comment]
Added a PR #12716 for the fix.
